src/Compilers/CSharp/Portable/Compilation/AttributeSemanticModel.cs

```diff
@@ -48,8 +48,29 @@ public static AttributeSemanticModel CreateSpeculative(SyntaxTreeSemanticModel p
             Debug.Assert(parentSemanticModel != null);
             Debug.Assert(rootBinder != null);
             Debug.Assert(rootBinder.IsSemanticModelBinder);
-
+            Debug.Assert(rootBinder.ContainingMemberOrLambda is not SourcePropertyAccessorSymbol { Property.IsIndexer: false } || hasSpeculativeFieldKeywordBinderInChain(rootBinder));
```

Comment:
This will need to traverse as well: consider an attribute on a local function, for instance.